File: RafJira/app/src/main/java/com/example/view/viewPager/PagerAdapterMainFragment.java
package com.example.view.viewPager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import com.example.view.fragments.NewTicketFragment;
import com.example.view.fragments.StatisticsFragment;
import com.example.view.fragments.TicketsListsFragment;
import com.example.view.fragments.UserAccountFragment;

public class PagerAdapterMainFragment extends FragmentPagerAdapter {

    private final int ITEM_COUNT = 4;
    public static final int FRAGMENT_1 = 0;
    public static final int FRAGMENT_2 = 1;
    public static final int FRAGMENT_3 = 2;
    public static final int FRAGMENT_4 = 3;

    public PagerAdapterMainFragment(@NonNull FragmentManager fm) {
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case FRAGMENT_1: return new StatisticsFragment();
            case FRAGMENT_2: return new NewTicketFragment();
            case FRAGMENT_3: return new TicketsListsFragment();
            default: return new UserAccountFragment();
        }
    }

    @Override
    public int getCount() {
        return ITEM_COUNT;
    }
    
}

File: NutritionTracker/app/src/main/java/com/example/nutritiontracker/presentation/navigation/viewPager/PagerAdapterMainFragment.kt
package com.example.nutritiontracker.presentation.navigation.viewPager

import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.example.nutritiontracker.presentation.fragments.CategoriesFragment
import com.example.nutritiontracker.presentation.fragments.LoginFragment

class PagerAdapterMainFragment(fm: FragmentManager) :  FragmentPagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    private val ITEM_COUNT = 4

    override fun getItem(position: Int): Fragment {
        Log.e("POSITION", position.toString())
        when(position){
            FRAGMENT_1 -> CategoriesFragment()
            FRAGMENT_2 -> {}
            FRAGMENT_3 -> {}
            FRAGMENT_4 -> {}
        }

        return CategoriesFragment()
    }

    override fun getCount(): Int {
        return ITEM_COUNT
    }

    companion object {
        const val FRAGMENT_1 = 0
        const val FRAGMENT_2 = 1
        const val FRAGMENT_3 = 2
        const val FRAGMENT_4 = 3
    }
}